modules/misc/xdg-autostart.nix

```diff
@@ -0,0 +1,40 @@
+{ config, lib, ... }:
+let
+  inherit (builtins) baseNameOf listToAttrs map unsafeDiscardStringContext;
+  inherit (lib) literalExpression mkEnableOption mkIf mkOption types;
+
+  cfg = config.xdg.autostart;
+
+  /* "/nix/store/x-foo/application.desktop" -> {
+       name = "autostart/application.desktop";
+       value = { source = "/nix/store/x-foo/application.desktop"; };
+     }
+  */
+  mapDesktopEntry = entry: {
+    name = "autostart/${unsafeDiscardStringContext (baseNameOf entry)}";
```

Comment:
Curious passer-by comment: why `unsafeDiscardStringContext`, especially as we have `value.source = entry` so it's going to be in the closure anyway?